src/c-writer.cc

```diff
@@ -4631,7 +4843,7 @@ void CWriter::WriteCHeader() {
   Write("#define ", guard, Newline());
   Write(Newline());
   Write(s_header_top);
-  Write(Newline());
+  ReserveExportNames();
```

Comment:
Should this happen before we even start writing the header?

Why remove the newline `Write(Newline());`

Reply:
In general we're getting double newlines in the generated output in a bunch of places, and this was one of them. It's tricky to handle systematically because some of the functions only write output if some nonobvious condition is true, so you don't want to unconditionally write a newline. I suspect what we may want is a new token and stream state that queues up a single extra Newline, but this can wait until later.

Reply:
(I restored all the old `Write(Newline())`s and will split that out to a later PR.)